www/src/utils/nav-items.js

```diff
@@ -0,0 +1,78 @@
+const _ = require(`lodash`)
+const { loadYaml } = require(`./load-yaml`)
+const navLinks = {
+  docs: loadYaml(`src/data/sidebars/doc-links.yaml`),
+  tutorial: loadYaml(`src/data/sidebars/tutorial-links.yaml`),
+  contributing: loadYaml(`src/data/sidebars/contributing-links.yaml`),
+}
+
+/**
+ * Flatten a nav hierarchy and append information about parents and children.
+ */
+function* flattenList(itemList, parents = []) {
```

Comment:
Why is this a generator function?

Reply:
It was just the cleanest way to express this IMO? I find generators are good at expressing complicated map/reduce expressions in a functional/efficient way. This is equivalent to

```
function flattenList(itemList, parents = []) {
  let items = []
  for (const item of itemList) {
    item.push({ …item /* other stuff */ })
    if (item.items) {
      item = item.concat(flattenList(item.items, [item.link, …parents])
    }
  }
  return items
}
```

I feel like that's harder to reason about since it's recursive?